### PR TITLE
CRM457-997: Manual assignment and unassignment

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
Manually assign unassigned applications to yourself
Unassign assigned applications from any caseworker

## Link to relevant ticket

[CRM457-997](https://dsdmoj.atlassian.net/browse/CRM457-997)

## How to manually test the feature

[CRM457-997]: https://dsdmoj.atlassian.net/browse/CRM457-997?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ivanELEC** [comment]
Potentially things that need to be taken out as separate bugs as I don't think this is technically jurisdiction of this ticket but there seems to be some weird goings on with assignment with certain cases. [This application](https://crm457-997-manual-nscc-caseworker-dev.cloud-platform.service.justice.gov.uk/prior_authority/applications/ac7aa34f-d34e-4643-834a-4a13970e22df) for example shows as "Not assigned" in the Open Applications page, even though it also shows that "case worker" is assigned to the application and you can't assign "super visor" to the application because there is already an assignment - other than that, this looks good

**Comment by patrick-laa** [comment]
@ivanELEC what apparently happened there was that the application had an assignment but the status was still 'submitted' rather than 'in_progress'. I think that may have been a dodgy data issue there. I've reset everything, done a bunch of assignments (manual and automatic) and unassignments, and it's not recurring, so I don't think the issue is in the application logic.

File: app/view_models/prior_authority/v1/application_summary.rb [hunk 1]
```diff
@@ -92,6 +92,14 @@ def current_section(current_user)
       def can_edit?(caseworker)
         submission.state == 'in_progress' && submission.assignments.find_by(user: caseworker)
       end
+
+      def unassignable?
+        submission.state == 'in_progress'
+      end
+
+      def assignable?
+        submission.state.in?(%w[submitted provider_updated])
```
Comment (dwhenry) [inline app/view_models/prior_authority/v1/application_summary.rb:101]: think long term I would prefer to have these as named things in a module..

i.e.
```
module States
  SUBMITTED = 'submitted'
  PROVIDER_UPDATED = 'provider_updated'

  ASSIGNABLE = [SUBMITTED, PROVIDER_UPDATED]
end
```

**Review by dwhenry — COMMENTED** [review]
LGTM

**Pull request merged**